app/services/iaa_reporting_helper.rb

```diff
@@ -37,4 +46,23 @@ def iaas
         end.compact
       end.sort_by(&:key)
   end
+
+  def partner_accounts
+    Agreements::PartnerAccount.
+      includes(integrations: :service_provider).
+      flat_map do |partner_account|
+        partner_account.integrations.map do |integration|
+          sp_issuer = integration.service_provider.issuer
+          if sp_issuer.present?
+            PartnerConfig.new(
+              partner_account_id: partner_account.id,
+              partner_agency: partner_account.requesting_agency,
```

Comment:
I think that we do not need this internal ID, the resulting report just needs the `requesting_agency` so I think we can use that here as the `partner`
```suggestion
              partner: partner_account.requesting_agency,
```

Reply:
yes, no need id,I will remove id.